plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -176,8 +184,13 @@ func defaultPredicates() sets.String {
 		factory.RegisterFitPredicate("CheckNodeDiskPressure", predicates.CheckNodeDiskPressurePredicate),
 
 		// Fit is determied by node condtions: not ready, network unavailable and out of disk.
-		factory.RegisterFitPredicate("CheckNodeCondition", predicates.CheckNodeConditionPredicate),
-		factory.RegisterMandatoryFitPredicate("CheckNodeCondition", predicates.CheckNodeConditionPredicate),
+		func() string {
+			// If taint node by condition is enabled, do not register node condition predicate.
+			if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) {
```

Comment:
Instead of inserting fake predicate you can conditionally insert `CheckNodeCondition` to `predSet` after it's constructed, before return.